fairlearn/datasets/_fetch_boston.py

```diff
@@ -108,15 +108,17 @@ def fetch_boston(
             Array of ordered feature names used in the dataset.
         DESCR : string
             Description of the Boston housing dataset.
-
-    (data, target) : tuple of (numpy.ndarray, numpy.ndarray)
-        if ``return_X_y`` is True and ``as_frame`` is False
-
-    (data, target) : tuple of (pandas.DataFrame, pandas.Series)
-        if ``return_X_y`` is True and ``as_frame`` is True
+        categories : dict or None
+            Maps each categorical feature name to a list of values, such that the
+            value encoded as i is ith in the list. If ``as_frame`` is True, this is None.
+        frame : pandas DataFrame
+            Only present when ``as_frame`` is True. DataFrame with ``data`` and ``target``.
+        
+    (data, target) : tuple if ``return_X_y`` is True
 
     Notes
-    -----
+    ----------
```

Comment:
```suggestion
    -----
```
the separator should have the same number of chars as the title above it.